modules/mobile_vr/register_types.cpp

```diff
@@ -37,7 +37,9 @@ void register_mobile_vr_types() {
 
 	Ref<MobileVRInterface> mobile_vr;
 	mobile_vr.instance();
-	XRServer::get_singleton()->add_interface(mobile_vr);
+	if (XRServer::get_singleton()) {
+		XRServer::get_singleton()->add_interface(mobile_vr);
```

Comment:
Do we risk leaking `mobile_vr` by not adding it as interface here? If so, maybe this should abort further up before instanciating it.

Reply:
I think not, but avoiding instantiation in the first place is a nice idea anyways, moved under condition. `ClassDB::register_class` is still needed for testing though.

It would be nice to merge #40920 to add those leak checks for that matter.